### ISSUE TITLE
autoadjusting number of characters when character combinations run out

### ISSUE THREAD

**Issue by NicoleSharp2000** [peterberbec/yourls-keyword_charset_length#4]
This plugin fails when the number of character combinations available runs out.  E.g. if using the default YOURLS 36-character set (leaving the custom charset blank in the plugin settings) and then setting the plugin to randomly generate 1-character links, the plugin will fail when the 37th link is created (since there are only 36 possible links using 1 character in a 36-character set).  While starting off with a larger number of characters for shortlinks will allow a larger number of possible combinations, in order for the plugin to be arbitrarily scalable, and also have the shortest possible links, it would be better to autoadjust the number of characters for shortlinks when the maximum number of link combinations is reached (i.e. increase the number of characters for shortlinks by one when there are no more shortlinks available).

As it is now, the plugin is appreciably well-behaved when the character combinations run out, and it is simply a matter of manually increasing the number of characters for shortlinks by one in order to continue being able to create random shortlinks.

**Comment by peterberbec** [comment]
I agree. I set myself to 6 character, so I am far from running out, but dynamically increasing the size would be best. I will look into this. If anyone has a PR, let me know!

Thanks for the idea!
